move the release scripts from `nushell/nu_scripts`

amtoine opened the pull request:
related to 
- https://github.com/nushell/nu_scripts/pull/572

# Description
this PR moves the *release* scripts from `nushell/nu_scripts`, removed in https://github.com/nushell/nu_scripts/pull/572, to `scripts/release/`.

this PR also fixes the readme and documentation of the scripts to mention `scripts/release/` instead of the old `make_release/` directory in the `nu_scripts` before https://github.com/nushell/nu_scripts/pull/572!

# User-Facing Changes

# Tests + Formatting

# After Submitting

fdncred commented:
`gen-js-ext.nu` doesn't need to be moved. it was for the vscode extension but i don't use it a lot anymore.

amtoine commented:
> `gen-js-ext.nu` doesn't need to be moved. it was for the vscode extension but i don't use it a lot anymore. and we don't have a clubhouse any longer so i'm not sure `20k_club.nu` needs to move.
>
gotcha :ok_hand: 

and **what about `get_coverage.nu` and `this_week_in_nu_weekly.nu`?**
they are not mentionned in the guide :thinking: 

we might either get rid of them or add them to the guide :yum:

fdncred commented:
`this_week_in_nu_weekly.nu` is used every week. It's what generates the post that Reilly or JT post from on formerly-known-as-Twitter. I figure it could stay where it is with the other ones. `get_coverage.nu` should probably stay put too, as it's not involved in releasing but may be used sometimes.

amtoine commented:
> `this_week_in_nu_weekly.nu` is used every week. It's what generates the post that Reilly or JT post from on formerly-known-as-Twitter. I figure it could stay where it is with the other ones.
>
ooh i see, i'll mention it in the guide then :ok_hand: 

> `get_coverage.nu` should probably stay put too, as it's not involved in releasing but may be used sometimes.
>
do you mean it should stay in here or moved back to `nu_scripts`?

fdncred commented:
> do you mean it should stay in here or moved back to nu_scripts?

I think both should probably stay in nu_scripts with the other ones i mentioned.

amtoine commented:
> I think both should probably stay in nu_scripts with the other ones i mentioned.
>
done in f7eb4421e :ok_hand: 

i've left `this_week_in_nu_weekly.nu` here though as the guide mentions tweeting in one of the last steps :yum: 
i've added a mention to the script in ada88fa36

fdncred commented on scripts/release/Readme.md:
```diff
@@ -0,0 +1,102 @@
+
+## 6. social media
+- [ ] post a status update on Discord
+- [ ] tweet about the new release using `./scripts/release/this_week_in_nu_weekly.nu`
```
this_week_in_nu_weekly.nu is not used during the release. It's just the script used for the weekly update post.

amtoine replied:
wow i completely misunderstood the goal of that script :scream: :laughing: 
let me correct this... :recycle:

amtoine replied:
done in 1234f6187

fdncred commented:
Release Post
![image](https://github.com/nushell/nushell/assets/343840/cbe64756-8c9e-42be-9ff7-ec5bec340138)

Weekly Post
![image](https://github.com/nushell/nushell/assets/343840/03daa5ff-e51d-4eec-a55f-abe4a112220c)

To be clear, I don't really care where the scripts go. I just want the instructions to be clear about what is used.

jntrnr reviewed (changes requested):
@amtoine - took a quick look. Generally looks okay, though I'd slim it down a lot.

A few of these scripts I would recommend against using. For example, the one that bumps the version will lose any comments and formatting in the files. 

get-full-changelog I think is similar to `since_last_release.nu` which... apparently I have locally but never got uploaded? oops. It's basically the same as the weekly thing but looks at a longer span. Though I think we're starting to use a different format based on the latest release, does this do that?

I guess I was surprised to see so many scripts. I only use `make_release.nu` and a modified version of "this week in nu" that grabs changes since the last release. Changing versions is a search/replace, tagging release does the work on github, and then we write the blog post. The template shouldn't live in this repo but instead in the website repo, if we want to keep it somewhere.

amtoine commented:
> I'd slim it down a lot.
> 
because i would not slim down the release guide, i think i would not slim down the scripts :thinking: 
however, as i say in the following, we can move more things around :yum: 

> A few of these scripts I would recommend against using. For example, the one that bumps the version will lose any comments and formatting in the files.
>
the `bump-version.nu` script is only a glorified find/replace, to make sure the person doing the bump doesn't do a mistake, where do you see it could lose commands and mess up the formatting in the files? :confused: 

> get-full-changelog I think is similar to `since_last_release.nu` which... apparently I have locally but never got uploaded? oops. It's basically the same as the weekly thing but looks at a longer span. Though I think we're starting to use a different format based on the latest release, does this do that?
>
`get-full-changelog` is a wrapper around `list-merged-prs` which does all the heavy lifting.
it might be similar but i would recommend using `get-full-changelog` instead of `this_week_in_nu_weekly.nu`, i'm not comfortable having my GitHub account password in environment variables, i'd rather use `gh` and be connected through the CLI tool :yum: 

> I guess I was surprised to see so many scripts. I only use `make_release.nu` and a modified version of "this week in nu" that grabs changes since the last release. Changing versions is a search/replace, tagging release does the work on github, and then we write the blog post.
>
all these scripts are part of the release guide and process :blush: 
i've written this guide as a sequence of safe steps to make sure the release goes well and no mistakes are being made, a bit like a safety sequence with tick-boxes for critical procedures!
to me it's not as trivial as pushing buttons, so i like having scripts and a complete guide to follow :relieved: 

> The template shouldn't live in this repo but instead in the website repo, if we want to keep it somewhere.
>
we could move the content of `./scripts/release/release-note/` to the website for instance?

### the general gist of this guide and scripts is to help maintainers like myself to do the release, relieving others from that burden :yum:

amtoine commented:
two changes not to forget about if we move these scripts there in the end
```diff
diff --git a/make_release/release-note/create-pr b/make_release/release-note/create-pr
index 3fe0716..de44bf3 100755
--- a/make_release/release-note/create-pr
+++ b/make_release/release-note/create-pr
@@ -48,7 +48,15 @@ def main [
 
     let title = $"Release notes for `($version)`"
     let body = $"Please add your new features and breaking changes to the release notes
-by opening PRs against the `release-notes-($version)` branch."
+by opening PRs against the `release-notes-($version)` branch.
+
+## TODO
+- [ ] look at interesting contributions
+- [ ] write all the sections
+- [ ] order the sections by interest
+- [ ] add the breaking changes
+- [ ] detail the breaking changes
+- [ ] add the full changelog"
 
     log info "creating release note from template"
     let release_note = $env.CURRENT_FILE
```
```diff
diff --git a/make_release/release-note/list-merged-prs b/make_release/release-note/list-merged-prs
index a6fed25..7c88584 100755
--- a/make_release/release-note/list-merged-prs
+++ b/make_release/release-note/list-merged-prs
@@ -1,4 +1,5 @@
 #!/usr/bin/env nu
+use std log
 
 def md-link [
     text: string
```

amtoine commented:
i'll go ahead and close this, the scripts have changed in the meantime and i'm not sure we have fully agreed to move them now :relieved:
